koku/koku/middleware.py

```diff
@@ -260,10 +262,38 @@ def create_customer(account, org_id, request_method):
                     customer.save()
                     UNIQUE_ACCOUNT_COUNTER.inc()
                     LOG.info("Created new customer from account_id %s and org_id %s.", account, org_id)
-        except IntegrityError:
-            customer = Customer.objects.filter(org_id=org_id).get()
+                    return customer
+                else:
+                    LOG.warning(
+                        log_json(
+                            msg="Customer object not saved",
+                            request_method=request_method,
+                            account_id=account,
+                            org_id=org_id,
+                        )
+                    )
+                    return None
 
-        return customer
+        except IntegrityError:
+            LOG.warning(
+                log_json(
+                    msg="IntegrityError when creating customer. Attempting to fetch existing record.",
+                    account_id=account,
+                    org_id=org_id,
+                )
+            )
+            try:
+                customer = Customer.objects.filter(org_id=org_id).get()
+                return customer
+            except Customer.DoesNotExist:
+                LOG.warning(
+                    log_json(
+                        msg="Customer does not exist. Returning None.",
+                        account_id=account,
+                        org_id=org_id,
+                    )
+                )
+                return None
```

Comment:
What is the impact of returning None? Won't this just error somewhere else?

Reply:
Yeah `None` maybe problematic else where like in some of the ProviderBuilder methods, returning an empty `Customer` object instead.

Reply:
I'm not sure this is a good idea either? How does this resolve the issue that we're encountering? What is the impact of passing around an empty Customer object??